server/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -155,6 +156,12 @@ public interface PluginServices {
          * The system indices for the cluster
          */
         SystemIndices systemIndices();
+
+        /**
+         * A service that resolves the data stream global retention that applies to
+         * data streams managed by the data stream lifecycle.
+         */
+        DataStreamGlobalRetentionResolver dataStreamGlobalRetentionResolver();
```

Comment:
I am not particularly happy for exposing this here since it's not widely used. But I didn't find another way to share it to the `data-streams` module. Please, let me know if there is a better way.

Reply:
Unfortunately not, if you can't do it through injection then it has to be through here.

Reply:
Thank you @thecoop